tests/cpp/test_scatter.cpp

```diff
@@ -168,7 +168,69 @@ TEST_F(ScatterTest, GridCounting) {
   testValidate(&fusion, outputs, {t0}, __LINE__, __FILE__);
 }
 
-TEST_F(ScatterTest, BlockCountingWithShmem2D) {
+TEST_P(ScatterTest, BlockCountingWithShmem2DExact) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  const std::vector<int64_t> self_shape{4, 100};
+  const std::vector<int64_t> index_shape{4, 10};
+
+  auto tv0 = makeContigConcreteTensor(index_shape, DataType::Int);
+  fusion.addInput(tv0);
+
+  auto tv1 = set(tv0);
+  auto tv2 = zeros(
+      {IrBuilder::create<Val>(self_shape[0]),
+       IrBuilder::create<Val>(self_shape[1])},
+      DataType::Int);
+  auto tv3 = ones(
+      {IrBuilder::create<Val>(index_shape[0]),
+       IrBuilder::create<Val>(index_shape[1])},
+      DataType::Int);
+  auto tv4 = scatter(tv2, 1, tv1, tv3);
+  auto tv5 = set(tv4);
+  fusion.addOutput(tv5);
+
+  auto options = at::TensorOptions().dtype(at::kLong).device(at::kCUDA, 0);
+  auto t0 = at::randperm(self_shape[1], options)
+                .slice(0, 0, index_shape[1])
+                .repeat({index_shape[0], 1});
```

Comment:
I'm repeating my earlier comment here. This example is ill-formed. We have many to one scatter.

Reply:
I think I took this pattern from somewhere related to MoE routing. Since the input is a permutation of integers, it should never hit the same position in the scatter output.

Reply:
You are right... I was reading the repeat here wrong... 🤕 
I thought we are repeating on the dimension of indexing.... Hence the suggestion to change the source tensor to be a non constant.